gui/src/2D/advancedDynamicTexture.ts

```diff
@@ -624,9 +656,51 @@ export class AdvancedDynamicTexture extends DynamicTexture {
             }
         });
 
+        this.onClipboardObservable.add((pi, state) => {
+            if (this.focusedControl) {
+                // call the event's callback
+                this.focusedControl.processKeyboard(pi.event);
+            }
+        });
+
         this._attachToOnPointerOut(scene);
     }
 
+    /** @hidden */
+    private onClipboardCopy = (evt: ClipboardEvent) => {
+        let ev = new ClipboardInfo(ClipboardEventTypes.COPY, evt);
+        this.onClipboardObservable.notifyObservers(ev);
+        evt.preventDefault();
+    }
+     /** @hidden */
+    private onClipboardCut = (evt: ClipboardEvent) => {
+        let ev = new ClipboardInfo(ClipboardEventTypes.CUT, evt);
+        this.onClipboardObservable.notifyObservers(ev);
+        evt.preventDefault();
+    }
+    /** @hidden */
+    private onClipboardPaste = (evt: ClipboardEvent) => {
+        let ev = new ClipboardInfo(ClipboardEventTypes.PASTE, evt);
+        this.onClipboardObservable.notifyObservers(ev);
+    }
+
+   /**
+    * Register the clipboard Events onto the canvas
+    */
+    public registerClipboardEvents(): void {

```

Comment:
register should be called somewhere during the ADT construction no?
Same for unregister, I would have called it during dispose

Reply:
I register them at onFocus() event of the input text and then deregister them at onBlur() 
I thought making them always listening to windows in not good. Imagine I am playing a game in which I have Ctr + C as walk fast/running or any other action, I wouldn't want copy event to be called.

Reply:
Well this makes sense actually :)